rmgpy/yml.py

```diff
@@ -0,0 +1,175 @@
+import yaml
+import os
+from rmgpy.chemkin import loadChemkinFile
+from rmgpy.species import Species
+from rmgpy.reaction import Reaction
+from rmgpy.thermo.nasa import NASAPolynomial, NASA
+from rmgpy.thermo.wilhoit import Wilhoit
+from rmgpy.kinetics.arrhenius import Arrhenius, PDepArrhenius, MultiArrhenius, MultiPDepArrhenius
+from rmgpy.kinetics.falloff import Troe, ThirdBody, Lindemann
+from rmgpy.kinetics.chebyshev import Chebyshev
+from rmgpy.data.solvation import SolventData
+from rmgpy.kinetics.surface import StickingCoefficient
+from rmgpy.util import makeOutputSubdirectory
+
+def convertChemkin2yml(chemkinPath, spcDictPath=None, output="chem.rms"):
+    if spcDictPath:
+        spcs,rxns = loadChemkinFile(chemkinPath,dictionaryPath=spcDictPath)
+    else:
+        spcs,rxns = loadChemkinFile(chemkinPath)
+    writeyml(spcs,rxns,path=output)
+
+def writeyml(spcs, rxns, solvent=None, solventData=None, path="chem.yml"):
+    D = getMechDict(spcs,rxns,solvent=solvent,solventData=solventData)
+    yaml.dump(D,stream=file(path,'w'))
+
+def getMechDict(spcs, rxns, solvent='solvent', solventData=None):
+    D = dict()
+    D["Units"] = dict()
+    D["Reactions"] = []
+    D["Phases"] = [dict()]
+    D["Phases"][0]["name"] = "phase"
+    D["Phases"][0]["Species"] = [obj2dict(x, spcs) for x in spcs]
+    D["Reactions"] = [obj2dict(x, spcs) for x in rxns]
+    if solventData:
+        D["Solvents"]= [obj2dict(solventData, spcs, label=solvent)]
+    return D
+
+def getradicals(spc):
+    if spc.molecule[0].toSMILES() == "[O][O]": #treat oxygen as stable to improve radical analysis
+        return 0
+    else:
+        return spc.molecule[0].multiplicity-1
+
+def obj2dict(obj, spcs, label="solvent"):
+    D = dict()
+    if isinstance(obj, Species):
+        D["name"] = obj.label
+        D["type"] = "Species"
+        D["smiles"] = obj.molecule[0].toSMILES()
+        D["thermo"] = obj2dict(obj.thermo, spcs)
+        if D["smiles"]  == "[O][O]":
```

Comment:
You defined the `getradicals` function above, so use this to get rid of the if-else statement here.

Reply:
done